WcaOnRails/app/views/regulations/scrambles/index.html.erb

```diff
@@ -71,5 +71,6 @@
     <li><%= link_to "TNoodle-WCA-0.13.5", "./tnoodle/old/TNoodle-WCA-0.13.5.jar" %> (2018-10-08)</li>
     <li><%= link_to "TNoodle-WCA-0.14.0", "./tnoodle/TNoodle-WCA-0.14.0.jar" %> (2018-12-03)</li>
     <li><%= link_to "TNoodle-WCA-0.15.0", "./tnoodle/TNoodle-WCA-0.15.0.jar" %> (2019-07-14)</li>
+    <li><%= link_to "TNoodle-WCA-0.15.1", "./tnoodle/TNoodle-WCA-0.15.1.jar" %> (2020-01-01)</li>
```

Comment:
Could you please create the pull request that introduces the tnoodle's jar and reference it here?
(you will have to change the base branch to "regulations-data" when creating the PR)

Reply:
> Could you please create the pull request that introduces the tnoodle's jar and reference it here?
> (you will have to change the base branch to "regulations-data" when creating the PR)

It's already been merged

https://github.com/thewca/worldcubeassociation.org/pull/5010

The only thing left is actually remove 0.15 from here

https://github.com/thewca/worldcubeassociation.org/tree/b7322a806e4190a48111a0da7114961a211b14a7/regulations/build/regulations/scrambles/tnoodle

Reply:
Btw, I kept both of them while deploying to staging without breaking production since both of them use the same branch.

Reply:
If you remove 0.15 from there then the link will be broken since it doesn't link to the "old" version (by the way the link to 0.14 is actually broken!).
The usual process for TNoodle in the past has been to create both merge request, and merge them at the same time (so that we don't have to add the jar twice and remove one of them).

But to be honest it feels quite awkward and as you noticed it's not quite handy to test changes (and it breaks existing links).
I don't remember why we were doing it that way, but maybe it's time to change this and just let old versions live next to new versions?

Instead of removing 0.15 from the tnoodle directory, could you actually:
  - remove 0.15 from "old"
  - move 0.14 from "old" to "tnoodle"

I believe this would restore the old broken link and make it easier for upcoming releases.

Reply:
Replying via email with lots of technical and practical reasons